src/package.ts

```diff
@@ -0,0 +1,57 @@
+import { privateEncrypt } from 'crypto';
+import contextKeys from './contextKeys';
+import { exec, pathExists } from './utilities';
+
+export class SPMPackage {
+	private constructor(
+        readonly folder: string,
+        public contents: any
+    ) {
+        this.setContextKeys()
+    }
+
+    public static async create(folder: string): Promise<SPMPackage> {
+        try {
+            let contents = await SPMPackage.loadPackage(folder)
+            return new SPMPackage(folder, contents)
+        } catch(error) {
+            // TODO: output errors
```

Comment:
Which errors are being caught here? It's not immediately clear (or documented) that `loadPackage` can throw.

I assume `exec` throws an error when `swift package describe --type json` fails to find `Package.swift`. Is that correct? If so, can we make this explicit to improve the clarity of the code?

Reply:
If you call `loadPackage` here and the `Package.swift` is invalid we should probably report the errors that are causing the issue. I'm guessing in the `Problems` section.

Reply:
I was just thinking that we should also handle cases where users are just editing Swift files, outside of a Swift package.  We should just disable the features that require a package then, but not throw any errors.

Reply:
But we should report errors which are errors in a Package.swift. Ok I should catch the no root manifest error and not report that

Reply:
Did you find a way to report errors in the `Problems` view? I cannot find any API to do that, only via problem matchers on tasks. The alternative would be to show an error notification, but then we'd have to make sure not to show it over and over again.

Reply:
Not yet was going to leave that for another PR